java/test/jmri/jmrix/ipocs/IpocsClientHandlerTest.java

```diff
@@ -0,0 +1,142 @@
+  }
+
+  @Test
+  public void completedClosedTest() throws IOException {
+    final IpocsClientHandler ch = new IpocsClientHandler(client);
+    ch.addClientListener(listener);
+    ch.completed(-1, null);
+    doThrow(new IOException()).when(client).close();
+    ch.completed(-1, null);
+  }
+
+  @Test
+  public void completedReceived1Test() throws IOException {
+    final IpocsClientHandler ch = new IpocsClientHandler(client);
+    ch.addClientListener(listener);
+    final Message msg = new Message();
+    msg.setObjectName("testing");
+    final ByteBuffer buff = msg.serialize();
+    buff.position(buff.capacity());
+    ch.completed(buff.capacity(), buff);
+  }
+
+  @Test
+  public void completedReceived2Test() throws IOException {
+    final IpocsClientHandler ch = new IpocsClientHandler(client);
+    ch.addClientListener(listener);
+    final Message msg = new Message();
+    msg.setObjectName("testing");
+    msg.getPackets().add(new SignOfLifePacket());
+    final ByteBuffer buff = msg.serialize();
+    buff.position(buff.capacity());
+    ch.completed(buff.capacity(), buff);
+  }
+
+  @Test
+  public void completedReceived3Test() throws IOException {
+    final IpocsClientHandler ch = new IpocsClientHandler(client);
+    ch.addClientListener(listener);
+    final Message msg = new Message();
+    msg.setObjectName("5");
+    final ConnectionRequestPacket pkt = new ConnectionRequestPacket();
+    pkt.setProtocolVersion((short) 0x0000);
+    pkt.setSiteDataVersion("3df43d");
+    msg.getPackets().add(pkt);
+    final ByteBuffer buff = msg.serialize();
+    buff.position(buff.capacity());
+    ch.completed(buff.capacity(), buff);
+  }
+
+  @Test
+  public void failedTest() throws IOException {
+    final IpocsClientHandler ch = new IpocsClientHandler(client);
+    ch.failed(new Exception("Error"), null);
+    jmri.util.JUnitAppender.suppressErrorMessage("Error closing connection");
+    ch.addClientListener(listener);
+    doThrow(new IOException()).when(client).close();
+    ch.failed(new Exception("Error"), null);
+    jmri.util.JUnitAppender.suppressErrorMessage("Error closing connection");
+  }
+
+  @Test
+  public void addClientListenerTest() {
+    final IpocsClientHandler ch = new IpocsClientHandler(client);
+    ch.addClientListener(listener);
+  }
+
+  @Test
+  public void removeClientListenerTest() {
+    final IpocsClientHandler ch = new IpocsClientHandler(client);
+    ch.removeClientListener(listener);
+  }
+
+  @Test
+  public void sendMessageTest() {
+    final IpocsClientHandler ch = new IpocsClientHandler(client);
+    final Message msg = mock(Message.class);
+    //final ArgumentCaptor<ByteBuffer> captor = ArgumentCaptor.forClass(ByteBuffer.class);
+    when(msg.serialize()).thenReturn(ByteBuffer.wrap(new byte[] {}));
+    //verify(client).write(captor.capture());
+    ch.send(msg);
+    //final ByteBuffer argument = captor.getValue();
+
+    //assertEquals(0, argument.capacity());
+  }
```

Comment:
What is the commented code for? Old code that isn't used any more or an unfinished test?

Reply:
thrown away code from when I was testing mocking and stuff. removed.